src/sensors/src/ModelSensorsTransformers.cpp

```diff
@@ -27,60 +27,152 @@ bool createReducedModelAndSensors(const Model& fullModel,
-                return false;
-            }
+                    // Get the name of the new first link (to which the old one has been lumped)
+                    std::string newFirstLinkName = reducedModel.getLinkName(newFirstLinkIndex);
 
-            iDynTree::LinkIndex secondLinkIndex = reducedModel.getLinkIndex(secondLinkName);
-            if( secondLinkIndex == iDynTree::LINK_INVALID_INDEX )
-            {
-                std::cerr << "[ERROR] createReducedModelAndSensors : " << secondLinkName << " is not in the reduced model, reducing sensors failed" << std::endl;
-                return false;
-            }
+                    // Set the name of the new firstLink and update its index
+                    sensorCopy->setFirstLinkName(newFirstLinkName);
+                    sensorCopy->updateIndices(reducedModel);
 
-            // Update indices
-            sensorCopy->updateIndices(reducedModel);
+                    // Update the transform
+                    sensorCopy->setFirstLinkSensorTransform(sensorCopy->getFirstLinkIndex(),
+                                                            newFirstLinkInReducedModel_H_oldFirstLinkInFullModel*oldFirstLinkInFullModel_H_sensorFrame);
 
-            reducedSensors.addSensor(*sensorCopy);
+                    // Update the appliedWrenchLink
+                    if (fullModel.getLinkName(sensorCopy->getAppliedWrenchLink()) == oldFirstLinkName) {
+                        sensorCopy->setAppliedWrenchLink(reducedModel.getLinkIndex(newFirstLinkName));
+                    }
+
+                    // Add the sensor to the reduced model
+                    reducedSensors.addSensor(*sensorCopy);
+                }
+                else {
+                    std::stringstream ss;
+                    ss << "The links related to the joint sensor attached on " << parentJointName << " have an invalid index" << std::endl;
+                    reportError("", "createReducedModelAndSensors", ss.str().c_str());
+                    delete sensorCopy;
+                    return false;
+                }
+
+                delete sensorCopy;
+            }
+            // If the joint to which is attached the sensor has been lumped, notify that the joint
+            // sensor will not be present in the reduced model
+            else {
+                std::stringstream ss;
+                ss << "The joint " << parentJointName << " is not in the reduced model, the associated joint sensor won't be present" << std::endl;
+                reportWarning("", "createReducedModelAndSensors", ss.str().c_str());
+
+            }
+        }
+        else {
+            std::stringstream ss;
+            ss << "The processed FT sensor couldn't be cast as a joint sensor" << std::endl;
+            reportWarning("", "createReducedModelAndSensors", ss.str().c_str());
 
-            delete sensorCopy;
         }
     }
 
-    // Then all link sensors
+    // Then all link sensors (ACCELEROMETERs and GYROSCOPEs)
```

Comment:
There are several types of link sensors, and in the future we can add more, and the code is already generic w.r.t. to that. I would avoid embedding this kind of comments that can become eventually updated.

Reply:
Ok